source/common/http/utility.cc

```diff
@@ -322,5 +322,32 @@ void Utility::appendToHeader(HeaderString& header, const std::string& data) {
   header.append(data.c_str(), data.size());
 }
 
+void Utility::extractHostPathFromUri(const std::string& uri, std::string& host, std::string& path) {
+  /**
+   *  URI RFC: https://www.ietf.org/rfc/rfc2396.txt
+   *
+   *  Example:
+   *  uri  = "https://example.com/certs"
+   *  pos:          ^
+   *  host_pos:       ^
+   *  path_pos:                  ^
+   *  host = "example.com"
+   *  path = "/certs"
+   */
```

Comment:
What about URIs with ports in them, e.g. http://foo.com:1234/foo? I think in general URIs get pretty complicated, for example you can encode usernames and passwords in them. This is kind of why I was in favor of having some standard library for this. If the limitations here are acceptable for your needs, that's fine, just document them.